scout/adapter/mongo/query.py

```diff
@@ -55,13 +55,15 @@ def build_query(self, case_id, query=None, variant_ids=None, category='snv'):
 
         # Requests to filter based on gene panels, hgnc_symbols or
         # coordinate ranges must always be honored. They are always added to
-        # query as top level, implicit '$and'.
+        # query as top level, implicit '$and'. When both hgnc_symbols and a 
+        # panel is used, addition of this is delayed until after the rest of 
+        # the query content is clear.
+ 
+        gene_query = []
 
         if query.get('hgnc_symbols') and query.get('gene_panels'):
-            gene_query = [
-                        {'hgnc_symbols': {'$in': query['hgnc_symbols']}},
-                        {'panels': {'$in': query['gene_panels']}}
-                    ]
+            gene_query.append({'hgnc_symbols': {'$in': query['hgnc_symbols']}})
+            gene_query.append({'panels': {'$in': query['gene_panels']}})
             mongo_query['$or']=gene_query
```

Comment:
Question: if you need to query by hgnc symbols AND panels, why there is OR in the query code?

Reply:
That semantics was never quite intuitive - but blame goes to the logicians and philosophers rather than me I think. If you want to use both a panel and add to it an additional free text set of symbols, OR is the operator to use. If we need both to be true at the same time, AND is correct.